### PR TITLE
Move MetadataId to the factory pattern.

### LINKED ISSUE #9045: MetadataId should use a Create factor method vs. a ctor
The `MetadataId` type should consider using a `Create` method over the current ctor to be consistent with other types in this area.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Don't have any public ways to create this type though.  There is no user
scenario where other clients would create instances of these.

Fixes #9045

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler  again. Sorry about the mixup.

File: src/Compilers/Core/Portable/MetadataReference/Metadata.cs [hunk 1]
```diff
@@ -14,6 +14,11 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public sealed class MetadataId
     {
+        private MetadataId()
+        {
+        }
+
+        internal static MetadataId CreateNewId() => new MetadataId();
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MetadataReference/Metadata.cs]: Why not simply "Create"?

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/Core/Portable/MetadataReference/Metadata.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Core/Portable/MetadataReference/Metadata.cs]: Good catch on making this `internal`

**Comment by jaredpar** [comment]
:+1:

File: src/Compilers/Core/Portable/MetadataReference/Metadata.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/MetadataReference/Metadata.cs]: @AlekseyTs Was just matching the pattern in the IDE code for our ID types.  As it's internal though, it shouldn't really matter one way or the other.  But i thought i'd at least be consistent there :)

**Comment by CyrusNajmabadi** [comment]
@MattGertz  For approval.

File: src/Compilers/Core/Portable/MetadataReference/Metadata.cs (hunk 1 above)
Comment (agocke) [inline src/Compilers/Core/Portable/MetadataReference/Metadata.cs]: +1 to `Create`. That's the pattern in the compiler code base (e.g., `Compilation.Create`)

Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/MetadataReference/Metadata.cs]: But does the compiler have other IDs it creates?  What's the pattern is uses for those?

**Comment by MattGertz** [comment]
Approved.

**Pull request merged**